rclrs/src/context.rs

```diff
@@ -7,13 +7,21 @@ use std::vec::Vec;
 use crate::rcl_bindings::*;
 use crate::{RclrsError, ToResult};
 
+/// This is locked whenever initializing or dropping any middleware entity
+/// because we have found issues in RCL and some RMW implementations that
+/// make it unsafe to simultaneously initialize and/or drop various types of
+/// entities. It seems these C and C++ based libraries will regularly use
```

Comment:
What types of entites do we need to be careful around besides logging (`rcl`) and middleware (`rmw`)? If it truly is all entities we can initialize/drop from our FFI deps, then perhaps we're can be more direct and say something to the effect of `... and/or drop entites from these libraries`

Reply:
As far as tests indicate, it's only what is mentioned here: middleware entities from RCL and RWM **implementation** (not `rmw` itself) libraries. There are some RCL data structures that we do not need to worry about, like `rcl_init_options_t` because its initialization and cleanup do not touch any global variables (..as far as I can tell).

I will probably participate in a discussion with rcl maintainers to improve the documentation around the use of global variables within rcl functions, and if that effort is fruitful then we'll be able to say with more confidence which exact functions to be concerned about.

Reply:
I've updated the documentation to clarify the full situation a bit: [0efa831](https://github.com/ros2-rust/ros2_rust/pull/386/commits/0efa8315d78baa1141551d2e433a58ff1a6717fb)